### DISCUSSION TITLE
Will there be a random map generator?

### DISCUSSION THREAD

**Discussion by DaisyBL** [ihhub/fheroes2#4651] [Ideas]
Hey,

You guys did a good job with the mod very impressed. My question is as above, will there ever be a rmg or is one planned for the future?
The random map generator that comes with the game inside the map editor is not really good when it comes to making maps sadly.

**Comment by ihhub** [comment]
Yes.

**Reply by Zyraxis** [reply]
Hi guys! 2 years have passed, and there is still no random map generator :( Is there at least an approximate date for its appearance? Is it possible to make a monetary contribution to the implementation of this particular task?

**Reply by ihhub** [reply]
Hi @Zyraxis , we are planning to release the Editor this year. Random map generator cannot be done without the Editor so the generator will be in 2024. There were a lot other very important issues which we had to fix and implement before starting work on the Editor. Also, we didn't have enough knowledge how the original map format works so it took significant amount of time to understand it.

Since this is an open source project where people contribute voluntary while they have free time, the development takes much longer than a normal fully funded commercial project with full-time employees :)

If you are willing to make a donation to the project you can read the information [here](https://github.com/ihhub/fheroes2#donation).

**Reply by oleg-derevenetz** [reply]
Or, if you can write high-quality code, just join the project to work on the editor and map generator :) Money by itself cannot replace good developers.
